plugins/modules/lxd_storage_pool_info.py

```diff
@@ -0,0 +1,342 @@
+
+- name: Get information about a specific storage pool
+  community.general.lxd_storage_pool_info:
+    name: default
+  register: result
+
+- name: Get information about all ZFS storage pools
+  community.general.lxd_storage_pool_info:
+    type: zfs
+  register: result
+
+- name: Get information about ZFS and BTRFS storage pools
+  community.general.lxd_storage_pool_info:
+    type:
+      - zfs
+      - btrfs
+  register: result
+
+- name: Get storage pool information via HTTPS connection
+  community.general.lxd_storage_pool_info:
+    url: https://127.0.0.1:8443
+    trust_password: mypassword
+    name: default
+  register: result
+
+- name: Get storage pool information for a specific project
+  community.general.lxd_storage_pool_info:
+    project: myproject
+  register: result
+"""
+
+RETURN = """
+storage_pools:
+  description: List of LXD storage pools.
+  returned: success
+  type: list
+  elements: dict
+  sample: [
+    {
+      "name": "default",
+      "driver": "dir",
+      "used_by": ["/1.0/instances/container1"],
+      "config": {
+        "source": "/var/lib/lxd/storage-pools/default"
+      },
+      "description": "Default storage pool",
+      "status": "Created",
+      "locations": ["none"]
+    }
+  ]
+  contains:
+    name:
+      description: The name of the storage pool.
+      type: str
+      returned: success
+    driver:
+      description: The storage pool driver/type.
+      type: str
+      returned: success
+    used_by:
+      description: List of resources using this storage pool.
+      type: list
+      elements: str
+      returned: success
+    config:
+      description: Configuration of the storage pool.
+      type: dict
+      returned: success
+    description:
+      description: Description of the storage pool.
+      type: str
+      returned: success
+    status:
+      description: Current status of the storage pool.
+      type: str
+      returned: success
+    locations:
+      description: Cluster member locations for the pool.
+      type: list
+      elements: str
+      returned: success
+logs:
+  description: The logs of requests and responses.
+  returned: when ansible-playbook is invoked with -vvvv.
+  type: list
+  elements: dict
+  sample: [
+    {
+      "type": "sent request",
+      "request": {
+        "method": "GET",
+        "url": "/1.0/storage-pools",
+        "json": null,
+        "timeout": null
+      },
+      "response": {
+        "json": {"type": "sync", "status": "Success"}
+      }
+    }
+  ]
+"""
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from ansible_collections.community.general.plugins.module_utils.lxd import (
+    LXDClient,
+    LXDClientException,
+    default_cert_file,
+    default_key_file,
+)
+
+# ANSIBLE_LXD_DEFAULT_URL is a default value of the lxd endpoint
+ANSIBLE_LXD_DEFAULT_URL = "unix:/var/lib/lxd/unix.socket"
+
+
+class LXDStoragePoolInfo(object):
```

Comment:
...and another Python 2 anachronism:
```suggestion
class LXDStoragePoolInfo:
```